server/index.js

```diff
@@ -216,6 +217,13 @@ export default class Server {
         }
 
         const p = join(this.dir, this.dist, 'bundles', 'pages', `${page}.js`)
+
+        // [production] If the page is not exists, we need to send a proper Next.js style 404
+        // Otherwise, it'll affect the multi-zones feature.
+        if (!(await fsAsync.exists(p))) {
```

Comment:
@arunoda so it still performs the server hit for non existing bundle, correct?

Reply:
For that yes. Basically you need to configure the CDN's to fallback to the server if it can't resolve.

I think that's totally doable for most of the CDNs with custom origin support.

Reply:
Many deployments use S3 for CDN origin, I think we're not the only company that does that. :)